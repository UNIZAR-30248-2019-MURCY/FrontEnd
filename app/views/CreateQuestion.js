import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    TextInput,
    ScrollView
} from 'react-native'
import {Button, CheckBox, Text} from 'react-native-elements';

export default class CreateQuestion extends Component {

    constructor(props) {
        super(props);
        this.state = {
            question: '',
            answer: '',
            checked: false,
            checkbox2: false,
            checkbox3: false,
            error: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if(this.state.question !== '' && this.state.answer !== ''){
            /*
            logInUser(this.state.email, this.state.password)
                .then((data) => {
                    console.log(data);
                    //this.setState(data);
                })
                .catch((error) => {
                    this.setState( {error: error.message})
                });
                */
               this.props.history.push('/editor');
        }
        else{
            this.setState({error: 'Introduzca todos los campos'})
        }

    }

    render() {
        return (
            <ScrollView>
            <View style={styles.container}>
                <View style={styles.containerMurcy}>
                    <Text h3>Create a new Question</Text>
                    <TextInput
                        style={styles.input}
                        placeholder='Question'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('question', val)}
                    />
                    <View>
                        <CheckBox 
                            style={styles.input}
                            center
                            title={<TextInput 
                                placeholder='Answer 1'
                                autoCapitalize="none"
                                placeholderTextColor='darkgrey'
                                onChangeText={val => this.onChangeText('answer', val)}/>}
                            checked={this.state.checked}
                            onPress={() => this.setState({ checked: !this.state.checked })}
                        />
                    </View>
                    <Button
                    buttonStyle={styles.button}
                    title="Create"
                    onPress={() => {
                        this.handleSubmit()
                    }}/>
                </View>
            </View>
            </ScrollView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerMurcy: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 160,
        padding: 20
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        marginTop: 70,
        width: 150,
        height: 55,
        padding: 8,
    }
})
