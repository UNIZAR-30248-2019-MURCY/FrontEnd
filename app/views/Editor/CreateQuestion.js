import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    TextInput,
    ScrollView
} from 'react-native'
import {Button, Text, CheckBox } from 'react-native-elements';


export default class CreateQuestion extends Component {

    constructor(props) {
        super(props);
        this.state = {
            title: '',
            description: '',
            options: [],

            title1: '',
            correct1: false,

            title2: '',
            correct2: false,

            title3: '',
            correct3: false,

            title4: '',
            correct4: false,

            error: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if(this.state.title !== '' && this.state.title1 !== '' && this.state.title2 !== ''){
            this.state.options.push({title : this.state.title1, correct  : this.state.correct1})
            this.state.options.push({title : this.state.title2, correct  : this.state.correct2})

            if(this.state.title3 !== ''){
                this.state.options.push({title : this.state.title3, correct  : this.state.correct3})
            }
            if(this.state.title4 !== ''){
                this.state.options.push({title : this.state.title4, correct  : this.state.correct4})
            }
            /*
            createQuestion(this.state.title, this.state.description, this.state.options)
                .then((data) => {
                    console.log(data);
                    //this.setState(data);
                })
                .catch((error) => {
                    this.setState( {error: error.message})
                });
                */
               this.props.navigation.replace('Editor');
        }
        else{
            this.setState({error: 'Enter title and minimum 2 answers'})
        }

    }

    render() {
        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );

        return (
            <ScrollView>
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2>New Question</Text>
                </View>
                <View style={styles.containerCreate}>
                    <TextInput
                        className='questionInput'
                        style={styles.input}
                        placeholder='Title'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('title', val)}
                    />
                    <TextInput
                        className='description'
                        style={styles.input}
                        placeholder='Description'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('description', val)}
                    />
                    <Text>--------------------------</Text>
                    <Text>Fill in at least 2 answers</Text> 
                    <View>
                        <TextInput
                            className='title1'
                            style={styles.input}
                            placeholder='Answer 1'
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('title1', val)}
                        />
                        <CheckBox value="1"
                        className='correct1' 
                        title='Mark as correct'
                        containerStyle={styles.checkBoxC}
                        checked={this.state.correct1}
                        onPress={() => this.setState({correct1: !this.state.correct1})} />
                        </View>
                        <View>
                        <TextInput
                            className='title2'
                            style={styles.input}
                            placeholder='Answer 2'
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('title2', val)}
                        />
                        <CheckBox value="2" 
                        className='correct2' 
                        title='Mark as correct'
                        containerStyle={styles.checkBoxC}
                        checked={this.state.correct2}
                        onPress={() => this.setState({correct2: !this.state.correct2})} />
                        </View>
                        <View>
                        <TextInput
                            className='title3'
                            style={styles.input}
                            placeholder='Answer 3'
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('title3', val)}
                        />
                        <CheckBox value="3"
                        className='correct3' 
                        title='Mark as correct'
                        containerStyle={styles.checkBoxC}
                        checked={this.state.correct3}
                        onPress={() => this.setState({correct3: !this.state.correct3})} />
                        </View>
                        <View>
                        <TextInput
                            className='title4'
                            style={styles.input}
                            placeholder='Answer 4'
                            placeholderTextColor='darkgrey'
                            onChangeText={val => this.onChangeText('title4', val)}
                        />
                        <CheckBox value="4" 
                        className='correct4' 
                        title='Mark as correct'
                        containerStyle={styles.checkBoxC}
                        checked={this.state.correct4}
                        onPress={() => this.setState({correct4: !this.state.correct4})} />
                    </View>
                    {showErr}
                    <Button
                    buttonStyle={styles.button}
                    className='create-button'
                    title="Create"
                    onPress={() => {
                        this.handleSubmit()
                    }}/>
                    <Button
                        className='return-button'
                        type="clear"
                        buttonStyle={styles.button2}
                        title="Return"
                        titleStyle={{color: 'grey'}}
                        onPress={() => {
                            this.props.navigation.goBack();
                        }
                        }/>
                </View>
            </View>
            </ScrollView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        marginTop: 50,
        padding: 20
    },
    containerCreate: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 10,
        padding: 20
    },
    checkBoxC: {
        backgroundColor: "white", 
        borderColor: "white"
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    }
})
