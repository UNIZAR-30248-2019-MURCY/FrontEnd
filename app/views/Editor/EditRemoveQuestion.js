import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    TextInput,
    ScrollView,
    FlatList,
    SafeAreaView, ActivityIndicator
} from 'react-native'
import {Button, Text, CheckBox, ListItem} from 'react-native-elements';

import {editQuestion, deleteQuestion, infoQuestion} from "../../services/quiz/questionFuncs";


export default class EditRemoveQuestion extends Component {

    constructor(props) {
        super(props);
        this.state = {
            data: [],
            loading: false,
            error: false,
        }
        this.onChangeText = this.onChangeText.bind(this);
    }

    componentDidMount() {
        if (this.props.navigation) {
            if(this.props.navigation.getParam('info')){
                this.setState({data: this.props.navigation.getParam('info')})
            }
        }
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    editQuestion() {
        this.setState({loading: true})
        this.setState({error: false})
        /*
        editQuestion(this.state.id, this.state.question, this.state.answer1, this.state.answer2, this.state.answer3, this.state.answer4, this.state.value)
            .then((data) => {
                console.log(data);
                this.setState({loading: false})
            })
            .catch((error) => {
                this.setState( {error: error.message})
                this.setState({loading: false})
            });
            */
        this.props.navigation.replace('QuestionConfirm', { type: 'modified' });
    }

    deleteQuestion() {
        this.setState({loading: true})
        this.setState({error: false})
        /*
        deleteQuestion(this.state.id)
            .then((data) => {
                console.log(data);
                this.setState({loading: false})
            })
            .catch((error) => {
                this.setState( {error: error.message})
                this.setState({loading: false})
            });
            */
        this.props.navigation.replace('QuestionConfirm', { type: 'deleted' });
    }

    render() {

        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]} className='loadingShow'>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        return (
            <ScrollView>
                <View style={styles.container}>
                    <View style={styles.containerTitle}>
                        <Text h2>Question</Text>
                    </View>
                    <View style={styles.containerCreate}>
                        <Text style={styles.subTitle}>Question and description</Text>
                        <TextInput
                            className='questionInput'
                            style={styles.input}
                            placeholder='Question'
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            value={this.state.data.title}
                            onChangeText={val => this.onChangeText('data.title', val)}
                        />
                        <TextInput
                            className='description'
                            style={styles.input}
                            placeholder='Description'
                            autoCapitalize="none"
                            placeholderTextColor='darkgrey'
                            value={this.state.data.description}
                            onChangeText={val => this.onChangeText('data.description', val)}
                        />

                        <Text style={styles.subTitle2}>Answers</Text>
                        <Text style={styles.subTitle3}>Fill in at least 2 answers</Text>

                        <FlatList
                            data={this.state.data.options}
                            keyExtractor={item => item.title}
                            renderItem={({item}) => (
                                <View style={styles.containerButtons}>
                                    <TextInput
                                        style={styles.inputAns}
                                        placeholder={item.title}
                                        placeholderTextColor='darkgrey'
                                        onChangeText={val => this.onChangeText('title1', val)}
                                    />
                                    <CheckBox value="1"
                                              containerStyle={styles.checkBoxC}
                                              checked={item.correct}
                                              onPress={() => this.setState({correct1: !this.state.correct1})}/>
                                </View>
                            )}
                        />

                        {showErr}
                        {showLoading}
                        <Button
                            className='edit-button'
                            buttonStyle={styles.button}
                            title="Edit"
                            onPress={() => {
                                this.editQuestion()
                            }}/>
                        <Button
                            className='delete-button'
                            buttonStyle={styles.button1}
                            title="Delete"
                            onPress={() => {
                                this.deleteQuestion()
                            }}/>
                        <Button
                            className='return-button'
                            type="clear"
                            buttonStyle={styles.button2}
                            title="Return"
                            titleStyle={{color: 'grey'}}
                            onPress={() => {
                                this.props.navigation.goBack();
                            }
                            }/>
                    </View>
                </View>
            </ScrollView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        //justifyContent: 'center',
        padding: 20
    },
    containerTitle: {
        flex: 1,
        alignItems: 'center',
        marginTop: 50,
        padding: 20
    },
    subTitle: {
        flex: 1,
        fontSize: 22,
        color: 'grey',
    },
    subTitle2: {
        flex: 1,
        fontSize: 22,
        color: 'grey',
        marginTop: 30,
    },
    subTitle3: {
        flex: 1,
        fontSize: 17,
        color: 'grey',
    },
    checkBoxC: {
        padding: 0,
        marginTop: 25,
        alignItems: 'center',
        backgroundColor: "white",
        borderColor: "white"
    },
    containerCreate: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    inputAns: {
        width: 260,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 50,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button1: {
        width: 120,
        height: 45,
        marginTop: 15,
        margin: 10,
        backgroundColor: 'red',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    containerLoading: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10
    },
    containerButtons: {
        flexDirection: 'row'
    },
})