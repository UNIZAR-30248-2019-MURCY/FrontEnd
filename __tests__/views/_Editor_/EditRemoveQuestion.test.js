import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import EditRemoveQuestion from "../../../app/views/Editor/EditRemoveQuestion";
import CreateQuestion from "../../../app/views/Editor/CreateQuestion";

describe('<EditRemoveQuestion />', () => {
    it('CreateQuestion renders without crashing', () => {
        const rendered = renderer.create(<EditRemoveQuestion/>).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('CreateQuestion test against snapshot', () => {
        const welcomePage = renderer.create(<EditRemoveQuestion/>).toJSON();
        expect(welcomePage).toMatchSnapshot();
    });
    it('handleSubmit: Edit Question', () => {
        let wrapper = shallow(<EditRemoveQuestion/>);
        wrapper.instance().editQuestion = jest.fn();
        wrapper.update();
        wrapper.find('.edit-button').simulate('press');
        expect(wrapper.instance().editQuestion).toBeCalledTimes(1);
    });
    it('handleSubmit: Delete Question', () => {
        let wrapper = shallow(<EditRemoveQuestion/>);
        wrapper.instance().deleteQuestion = jest.fn();
        wrapper.update();
        wrapper.find('.delete-button').simulate('press');
        expect(wrapper.instance().deleteQuestion).toBeCalledTimes(1);
    });
    it('goBack and navigate to QuestionsEdit', () => {
        const navigationMock = {
            goBack: jest.fn(),
            getParam: jest.fn(),
        };
        const wrapper = shallow(<EditRemoveQuestion navigation={navigationMock}/>);
        wrapper.find('.return-button').simulate('press');
        expect(navigationMock.goBack).toHaveBeenCalled()
    });
    it('should render the Error component if state.error is true', () => {
        let createComponent = shallow(<EditRemoveQuestion/>);
        createComponent.setState({error: true});
        expect(createComponent.find('.errorShow').length).toBe(1);
    });
    it('should render the Loading component if state.loading is true', () => {
        let createComponent = shallow(<EditRemoveQuestion/>);
        createComponent.setState({loading: true});
        expect(createComponent.find('.loadingShow').length).toBe(1);
    });
    it('onChangeText', () => {
        let wrapper = shallow(<EditRemoveQuestion/>);
        wrapper.instance().onChangeText = jest.fn();

        let event = 'What is the capital of Spain?';
        wrapper.find('.questionInput').simulate('changeText', event)
        expect(wrapper.instance().onChangeText).toBeCalledWith('title','What is the capital of Spain?');

        event = 'Description';
        wrapper.find('.description').simulate('changeText', event)
        expect(wrapper.instance().onChangeText).toBeCalledWith('description','Description');

    });
});
