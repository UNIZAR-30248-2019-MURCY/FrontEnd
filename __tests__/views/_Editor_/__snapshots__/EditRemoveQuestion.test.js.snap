// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRemoveQuestion /> CreateQuestion test against snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": 50,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 42.5,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Question
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "flex": 1,
              "fontSize": 22,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Question and description
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          className="questionInput"
          onChangeText={[Function]}
          placeholder="Question"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          className="description"
          onChangeText={[Function]}
          placeholder="Description"
          placeholderTextColor="darkgrey"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "whitesmoke",
              "borderRadius": 14,
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "height": 55,
              "margin": 10,
              "padding": 8,
              "width": 300,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Text
          style={
            Object {
              "color": "grey",
              "flex": 1,
              "fontSize": 22,
              "marginTop": 30,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Answers
        </Text>
        <Text
          style={
            Object {
              "color": "grey",
              "flex": 1,
              "fontSize": 17,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Fill in at least 2 answers
        </Text>
        <RCTScrollView
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
        <View />
        <View />
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "grey",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 55,
                  "justifyContent": "center",
                  "margin": 10,
                  "marginTop": 50,
                  "padding": 8,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Edit
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "margin": 10,
                  "marginTop": 15,
                  "padding": 8,
                  "width": 120,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 14,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 55,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Return
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
